actions/promote/promoteBatch.js

```diff
@@ -0,0 +1,258 @@
+/* ************************************************************************
+* ADOBE CONFIDENTIAL
+* ___________________
+*
+* Copyright 2023 Adobe
+* All Rights Reserved.
+*
+* NOTICE: All information contained herein is, and remains
+* the property of Adobe and its suppliers, if any. The intellectual
+* and technical concepts contained herein are proprietary to Adobe
+* and its suppliers and are protected by all applicable intellectual
+* property laws, including trade secret and copyright laws.
+* Dissemination of this information or reproduction of this material
+* is strictly forbidden unless prior written permission is obtained
+* from Adobe.
+************************************************************************* */
+
+const openwhisk = require('openwhisk');
+const filesLib = require('@adobe/aio-lib-files');
+const { getConfig } = require('../config');
+const {
+    getAuthorizedRequestOption, fetchWithRetry
+} = require('../sharepoint');
+const {
+    getAioLogger, logMemUsage, delay, PROMOTE_ACTION
+} = require('../utils');
+const appConfig = require('../appConfig');
+const urlInfo = require('../urlInfo');
+const FgStatus = require('../fgStatus');
+const BatchManager = require('../batchManager');
+
+const logger = getAioLogger();
+const MAX_CHILDREN = 5000;
+
+/**
+ * This is action interfacing method. The worker is split into two promoteBatch and worker.
+ * This promoteBatch has following
+ * 1. Searches for the files to be promoted (using msal search api)
+ * 2. Using the batchmanger adds the files and this batchmanager splits into batches.
+ * The batch information is also stored in manifest files and the actions are triggered
+ * Following parameters are used and needs to be tweaked
+ * 1. Number of files per batch - Assume 10k
+ * 2. Number of activation per container - Assume 5
+ * 3. Number of parallel copies - Assume 20
+ * Assuming avg 5MB/file and assume that all parallel copies are loaded (i.e 20) total size is 100mb
+ * and 5 container total is about 0.5gb and below treshold and can be tweaked.
+ */
+async function main(params) {
+    const ow = openwhisk();
+    logMemUsage();
+    let payload;
+    const {
+        adminPageUri, projectExcelPath, fgRootFolder
+    } = params;
+    appConfig.setAppConfig(params);
+    const fgStatus = new FgStatus({ action: PROMOTE_ACTION, statusKey: fgRootFolder });
+    const filesSdk = await filesLib.init();
+    const batchManager = new BatchManager({ action: PROMOTE_ACTION, filesSdk });
+    // For current cleanup files before starting
+    await batchManager.cleanupFiles();
```

Comment:
We should move this into a separate action so that it does not slow down the promote job, especially considering that this is only required for testing purposes (and perhaps for eventual files clean up when used on live content which can still be run separately and need not be part of the promote action).